freshmaker/pyxis.py

```diff
@@ -0,0 +1,205 @@
+
+import requests
+import logging
+import json
+from http import HTTPStatus
+from requests_kerberos import HTTPKerberosAuth, OPTIONAL
+from packaging import version
+from math import ceil
+
+from freshmaker import log, conf
+
+# decrease level of logging for kerberos authentication
+logging.getLogger('requests_kerberos').setLevel('INFO')
+
+
+class PyxisRequestError(Exception):
+    """
+    Error return as a response from Pyxis
+    """
+
+    def __init__(self, status_code, error_response):
+        """
+        Initialize Pyxis request error
+
+        :param int status_code: response status code
+        :param str or dict error_response: response content returned from Pyxis
+        """
+
+        self._status_code = status_code
+        self._raw = error_response
+
+    @property
+    def raw(self):
+        return self._raw
+
+    @property
+    def status_code(self):
+        return self._status_code
+
+    def __str__(self):
+        return 'Errors ({}):\n{}'.format(
+            len(self._raw['errors']),
+            '\n'.join(('    {}'.format(err['msg'])
+                      for err in self._raw['errors']))
+        )
+
+
+class Pyxis(object):
+    """ Interface for querying Pyxis"""
+
+    def __init__(self, server_url):
+        self._server_url = server_url.rstrip('/')
+        # add api version to root url
+        self._api_root = f"{self._server_url}/v1"
```

Comment:
Optional: This is more Pythonic:
```suggestion
        self._api_root = urllib.parse.urljoin(self._server_url, "v1")
```